gateway/ocfl_repository_gateway.py

```diff
@@ -51,29 +50,33 @@ def create_staged_object(self, id: str) -> None:
                 raise StagedObjectAlreadyExistsError() from e
             raise RepositoryGatewayError() from e
 
-    def stage_object_files(self, id: str, source_package: Package) -> None:
-        root_path = source_package.get_root_path()
-        full_child_paths = ((root_path / child) for child in source_package.get_children())
+    def _stage_object_file(self, id, source_path: Path, dest_path: Path) -> None:
         command = [
             "rocfl", "-r", str(self.storage_path),
-            "cp", "-r", id, *full_child_paths, "--", "/"
+            "cp", "-r", id, str(source_path.absolute()), "--", str(dest_path)
         ]
         try:
             subprocess.run(command, check=True, capture_output=True)
         except CalledProcessError as e:
-            not_found_message = f"Not found: Object {id}"
-            if not_found_message in e.stderr.decode():
-                raise StagedObjectDoesNotExistError() from e
             raise RepositoryGatewayError() from e
 
+    def stage_object_files(self, id: str, source_package: Package) -> None:
+        if not self.has_object(id) and not self._has_staged_object(id):
+            raise ObjectDoesNotExistError(f"No object or staged object found for id {id}")
```

Comment:
This was the only usage of `StagedObjectDoesNotExistError`, but our switch to preemptively checking for whether `id` is valid (i.e. either refers to an existing versioned object or a staged object) removed it. I elected to raise`ObjectDoesNotExistError` with a string message in the same way `get_object_files` does.